cms/src/enveloped_data.rs

```diff
@@ -51,6 +52,24 @@ pub struct EnvelopedData {
     pub unprotected_attrs: Option<Attributes>,
 }
 
+impl<'__der_lifetime> ::der::DecodeValue<'__der_lifetime> for Box<EnvelopedData> {
```

Comment:
Can you change the lifetimes to `'a` and add a TODO comment:

```suggestion
// TODO: replace these handwritten impls with `Box` support in the proc macro
impl<'a> ::der::DecodeValue<'a> for Box<EnvelopedData> {
```

Reply:
Done. There is a draft macro with box support and one struct changed to use it on the box branch of my fork.